mealie/repos/repository_recipes.py

```diff
@@ -151,29 +152,95 @@ def _uuids_for_items(self, items: list[UUID | str] | None, model: type[SqlAlchem
         return ids + additional_ids
 
     def _add_search_to_query(self, query: Select, search: str) -> Select:
+        """
+        0. fuzzy search (postgres only) and tokenized search are performed separately
+        1. take search string and do a little pre-normalization
+        2. look for internal quoted strings and keep them together as "literal" parts of the search
+        3. remove special characters from each non-literal search string
+        4. token search looks for any individual exact hit in name, description, and ingredients
+        5. fuzzy search looks for trigram hits in name, description, and ingredients
+        6. Sort order is determined by closeness to the recipe name
+        Should search also look at tags?
+        """
+
         normalized_search = unidecode(search).lower().strip()
+        punctuation = "!\#$%&()*+,-./:;<=>?@[\\]^_`{|}~"  # string.punctuation with ' & " removed
+        # keep quoted phrases together as literal portions of the search string
+        literal = False
+        quoted_regex = re.compile(r"""(["'])(?:(?=(\\?))\2.)*?\1""")  # thank you stack exchange!
```

Comment:
Let's precompile these so we don't take the perf hit on each search

Reply:
Not sure I understand. Quoted_regex is already precompiled, and punctuation is only used in str.maketrans

Reply:
You're still compiling per request instead of at startup, it's fine though it likely doesn't make that big of a difference